### PR TITLE
dev/core#3458 Fix table and alias for Event ActionMapping check

### PR DESCRIPTION
**Pull request by agileware-fj**
Overview
----------------------------------------
Fixes an issue where scheduled reminders fail for **Event** reminders with:
- Repetition turned on, and
- "Limit or Add Recipients" is set to "Also Include"

Due to a DB Error: `Not unique table/alias: 'r'`

Before
----------------------------------------
CRM_Event_ActionMapping has the `casAddlCheckFrom` query parameter set to `civicrm_event r`, and also specifies a join to the `civicrm_event` table with the same alias.
Not only is the alias duplicate (and incorrect according to the later query), but the entity appears wrong.

The query generated to check if there are any recipients in the relevant phase to look like:

```sql
SELECT *
FROM civicrm_event r
INNER JOIN civicrm_event r ON e.event_id = r.id
INNER JOIN civicrm_contact c ON c.id = e.contact_id AND c.is_deleted = 0 AND c.is_deceased = 0
WHERE (r.id IN ("111")) AND (r.is_active = 1) AND (r.is_template = 0) AND ("20220512113156" <= DATE_SUB(
start_date, INTERVAL 3 hour))
LIMIT 1
OFFSET 0
```

Exception is thrown due to DB Error, further scheduled reminders are not sent (cron job does not appear to stop, though)

After
----------------------------------------
Replaces the incorrect table / alias with `civicrm_participant e`

Resulting query:

```sql
SELECT *
FROM civicrm_participant e
INNER JOIN civicrm_event r ON e.event_id = r.id
INNER JOIN civicrm_contact c ON c.id = e.contact_id AND c.is_deleted = 0 AND c.is_deceased = 0
WHERE (r.id IN ("111")) AND (r.is_active = 1) AND (r.is_template = 0) AND ("20220512113156" <= DATE_SUB(
start_date, INTERVAL 3 hour))
LIMIT 1
OFFSET 0
```

Exception is not thrown due to DB Error, Scheduled Reminders are sent

Technical Details
----------------------------------------

The `e` alias is in line with other ActionMapping classes, and lines up with the reset of the generated query.
`civicrm_participant` is the correct table to associate a Contact with with an event, which appears to be what we're checking for – i.e. only send reminders to additional recipients for events that have participants.

Other
----------------------------------------
Agileware ref CIVICRM-1981

### CODE REVIEW

**Comment by MegaphoneJon** [comment]
Note that this configuration has been disabled in Civi 5.49 - it has far more problems than a crash.  E.g. the "also include" group will receive the email almost immediately after it's created, not when it's scheduled.

The problem stems from
References: 
https://github.com/civicrm/civicrm-core/pull/23101
https://github.com/civicrm/civicrm-core/pull/17641#issuecomment-664094558

**Comment by agileware-fj** [comment]
@MegaphoneJon You can still do it from the "Schedule Reminders" tab in Event Management though... this is how I tested on master.

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon are the problems you mention tied up with the regression we put out a fix for?

**Comment by kcristiano** [comment]
I was able to reproduce this on a site I just upgraded to 5.49.3.

This site had scheduled reminders that had been set some time ago so the changes [mentioned here ](https://github.com/civicrm/civicrm-core/pull/23445#issuecomment-1124549038) were already in place.

Applying the PR fixed the issue.

**Comment by totten** [comment]
So... this seems to be a vampire configuration? Neither alive nor dead?

* To bring it back to life, we would want a fix (like this) and more test-coverage -- and then re-enable in web UI.
* To put it to eternal slumber, we would want to update/migrate any existing records and disable in any remaining UIs.

Some thoughts:

* Skimming around, it does appear that this value of `casAddlCheckFrom` is more consistent with the other ways that `casAddlCheckFrom` is used. (*Ex: `casAddlCheckFrom` can also be used with `civicrm_contribution` and `civicrm_activity` but not `civicrm_contribution_page`.*) And @kcristiano says it provides a fix for him. So it seems pretty plausible.
* According to `git tag --contains` and `git branch --contains`, the change in #23101 actually takes effect in `5.50-rc`. It's not in `5.49-stable`.
* I don't have an opinion yet about whether this configuration should be revived, killed, or left as a vampire. There's a preliminary question for which I don't have an answer: _How did it behave in the past, and how was that good/bad?_
* As a general matter, I tend to agree with @MegaphoneJon's POV in #23101 -- the "Also Include" functionality feels confusing (at a design level).
* I looked for unit-tests to explain the expected behavior. There's not a lot. I found an interesting bit in the test-suite for Contributions+"Also Include", played with it, and submitted more comments/explanations in #23628.
* Part of me suspects that this should be replaced by other functionality (eg a "CC" feature and a "Preview"/"Simulator" feature). But those are obviously tasks that wouldn't fit in 5.49/5.50.

**Comment by agileware-fj** [comment]
> So... this seems to be a vampire configuration? Neither alive nor dead?

We only came across it because of ... Upgrading sites to 5.49 that already had this configuration set.

It was definitely a lot more obvious with the addition of the scheduled jobs failure check, but I'm not quite sure why it wasn't making the whole thing not finish beforehand - since when I looked the erroneous code appears to have been there since the dawn of git.

This configuration set doesn't make a whole lot of sense, as noted; this PR only exists because we just needed to get the jobs running and felt that fixing the bug was less dangerous than assuming that no-one was actually using it on purpose.  Although on the flip side, this would also mean that more emails could be randomly and nonsensically sent to groups of people they were never intended for.

In summary – it sounds then like the better option is to put a lemon in its mouth and cut the head off by way of an upgrade script that removes such nonsensical configurations.

**Comment by eileenmcnaughton** [comment]
Reading the above it seems like this should be closed & maybe another upgrade script one added

**Comment by MegaphoneJon** [comment]
I don't have access right now, but when I was working on disabling this 
configuration, I recall that previously that a "0" had a different 
meaning depending on whether group_id was populated.Message ID: 
***@***.***>

**Comment by agileware-fj** [comment]
This may have been fixed by 83ef8f4 ... either way, the PR doesn't apply any more.
